MarvelCharsApp/MarvelCharsApp/Views/SearchTableViewController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  SearchTableViewController.swift
+//  MarvelCharsApp
+//
+//  Created by User on 11/25/21.
+//
+
+import UIKit
+
+class SearchTableViewController: UITableViewController {
+    
+    let searchVm = SearchViewModel()
+    let searchController = UISearchController(searchResultsController: nil)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        initialConfiguration()
+    }
+    
+    //Hide Navigation bar when view appears and show it when it disappear
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+    
+    private func initialConfiguration() {
+        //Search UI configurations
+        searchController.searchResultsUpdater = self
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.searchBar.placeholder = "Search Characters"
+        searchController.searchBar.delegate = self;
+        tableView.tableHeaderView = searchController.searchBar
+        definesPresentationContext = true
+        // Search VM configurations
+        searchVm.getAllcharacters()
+        assignClosureToViewModel()
+    }
+    
+    private func assignClosureToViewModel() {
+        searchVm.refreshData = {
+            DispatchQueue.main.async { [weak self] in
+                self?.tableView.reloadData()
+            }
+        }
+    }
+    
+    // MARK: - Table view data sourc
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchVm.count()
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "nameCell", for: indexPath)
```

Comment:
Use identifiers as constants, not just strings

Reply:
I did a static class to access to identifiers directly